pkg/codegen/internal/test/sdk_driver.go

```diff
@@ -29,14 +29,68 @@ type sdkTest struct {
 	// are of the form "$language/$check" such as "go/compile".
 	Checks map[string]CodegenCheck
 
-	// Skip checks, identified by "$language/$check".
-	// "$language/any" is special, skipping generating the
+	// KnownBroken indicates tests that are known to be failing, identified by
+	// "$language/$check". "$language/any" is special, skipping generating the
 	// code as well as all tests.
-	Skip codegen.StringSet
+	KnownBroken codegen.StringSet
 
 	// Do not compile the generated code for the languages in this set.
-	// This is a helper form of `Skip`.
-	SkipCompileCheck codegen.StringSet
+	// This is a helper form of KnownBroken.
+	KnownBrokenCompile codegen.StringSet
+
+	// IntendedFor specifies languages[/tests] that this test is intended to run
+	// for. Unlike KnownBroken, IntendedFor indicates that the test should only
+	// run for the specified input. It is not used to indicate that other tests
+	// are broken.
+	IntendedFor codegen.StringSet
```

Comment:
Bit sad having two properties that both result in skipping and nothing to force them to be in sync. Could we not have just set Skip with something like "allLanguages.Subtract("dotnet")"?

Reply:
I have reverted back to `Skip` and added the `allLanguages` variable (good idea!). My intent wasn't to provide an inverse to `Skip`, it was to communicate that a test had an intended audience, distinct from just being known to fail.